ctrl/ctrl.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2025 Pantacor Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef PV_CTRL_H
+#define PV_CTRL_H
+
+#include "ctrl_callback.h"
+
+#include <stdbool.h>
+
+struct evhttp_request;
+
+int pv_ctrl_start(void);
+void pv_ctrl_stop(void);
+int pv_ctrl_add_endpoint(const char *path, const int methods, bool mgmt,
```

Comment:
IMO, pv_ctrl_add_endpoint and most of the code in ctrl.c and ctrl_callback.c could go to somehow to event/ and leave ctrl/ for the policy implementation specific to pv-ctrl (*ep.c files)

Reply:
Not sure about it, maybe we need to generalize more, for example, if ctrl would has a parameter to set the unix socket.
In my opinion, we should do it if pantavisor needed another http server, and it feels like `event/` isn't the right place, maybe a new folder like `http/` or `websrv/`